ci/ci/github.py

```diff
@@ -456,6 +457,29 @@ def checkout_script(self):
 git merge {shq(self.source_sha)} -m 'merge PR'
 '''
 
+    def checkout_job(self, batch):
+        if batch.id not in self.jobs:
+            script = f'''
+set -ex
+date
+
+cd /io/
+rm -rf repo
+mkdir repo
+cd repo
+{self.checkout_script()}
```

Comment:
Instead of hard-coding the clone step, let's just have it be an explicit clone step in the build.yaml.  The build step already supports input files, so it can just grab the relevant files as needed (possibly with some path changes to set the right docker context).  I think that makes the CI code simpler and gives us more control in the build.  (Eventually, if we wanted the old functionality back, we could add a clone option to docker build.)